sqlglot/dialects/bigquery.py

```diff
@@ -778,6 +780,11 @@ class Generator(generator.Generator):
             "within",
         }
 
+        def table_sql(self, expression: exp.Table, sep: str = " AS ") -> str:
+            if expression.meta.get("quoted_table"):
```

Comment:
can we do this one without mutation? since it's so common

expression = exp.Table(this=exp.Identifier(this=p.naem for ...))

Reply:
What's the advantage of constructing a new `Table` vs mutating here? Remember that we'll also have to carry over the other args in the table.

Reply:
Whoops, good catch yeah - I intended for this to be an identifier.

> what do you mean you have to carry over args?

I mean that you can't just do `expression = exp.Table(this=...)` - what if `expression` had more [args](https://github.com/tobymao/sqlglot/blob/17e34e79d22e3c8211f1bf42047d4ed3557628b6/sqlglot/expressions.py#L2620)? With this overwrite here you'll drop them.

Reply:
https://github.com/tobymao/sqlglot/pull/3021/commits/a9ac1c21e690ed6d3c3c0087542bf769677f0306